### Problem Statement 1

##### H0: mu =100
##### H1: mu!=100

In [1]:
pop_mean = 100
pop_std = 12
n = 36
sam_mean = 108

In [2]:
import numpy as np
from scipy import stats

In [3]:
def get_zscore_1(mu,sigma,x_bar,n):
    return (x_bar-mu)/(sigma/np.sqrt(n))

In [4]:
z_score = get_zscore_1(pop_mean,pop_std,sam_mean,n)
z_score

4.0

In [5]:
alpha = 0.05
p = 1-alpha/2

critical=stats.norm.ppf(abs(p))
critical


1.959963984540054

In [6]:
if(abs(z_score)>critical):
    print("H0 rejected")
else:
    print("H0 accepted")

H0 rejected


### Problem Statement 2

#### H0 : p_1 >= p_2
#### H1 : p_1<p_2 

In [7]:
def get_zscore(p_1,p_2,n1,n2,D=0): #D = difference in population proportion
    numerator = (p_1-p_2)-D
    denominator = np.sqrt(p_1*(1-p_1)/n1 +p_2*(1-p_2)/n2)
    z_value = numerator/denominator
    return z_value
    

In [8]:
p_1 = 0.52
p_2 = 0.47
n1 = 100
n2=100
z_score = get_zscore(p_1,p_2,n1,n2)

In [9]:
z_score

0.7080278164104213

In [10]:
p_value=1 - stats.norm.cdf(abs(z_score))
print("Probability of greater percentage of Republican voters in second state than in first state: {}".format(p_value))

Probability of greater percentage of Republican voters in second state than in first state: 0.23946399182220013


### Problem Statement 3

In [13]:
z_score = get_zscore_1(1026,209,1100,n=1)
z_score

0.35406698564593303

#### I scored 0.35 standard deviation time above than the average test taker